test/extended/oauth/groupsync.go

```diff
@@ -1,110 +1,419 @@
+			o.Expect(err).NotTo(o.HaveOccurred(), "output: %s", out)
+
+			o.Expect(err).NotTo(o.HaveOccurred(), "output: %s", out)
+			_, err = oc.Run("adm", "groups").Args("prune", "--sync-config="+syncConfigPath, "--confirm").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			compareAndCleanup(oc, path.Join(currentDir, "valid_all_ldap_sync_prune.yaml"))
+
+			g.By("Sync all LDAP groups from LDAP server using whitelist file, then prune OpenShift groups using the same whitelist file")
+			out, err = oc.Run("adm", "groups", "sync").Args("--whitelist="+path.Join(currentDir, "ldapgroupuids.txt"), "--sync-config="+syncConfigUserDefinedPath, "--confirm").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).To(o.ContainSubstring("group/"))
+			checkOnlyGroupsExist(userClient, "firstgroup", "secondgroup", "thirdgroup")
+
+			out, _, err = oc.Run("adm", "groups").Args("prune", "--whitelist="+path.Join(currentDir, "ldapgroupuids.txt"), "--sync-config="+syncConfigUserDefinedPath, "--confirm").Outputs()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).To(o.Equal(""))
+			checkOnlyGroupsExist(userClient, "firstgroup", "secondgroup", "thirdgroup")
+
+			out, err = oc.Run("patch").Args("group", "secondgroup", "-p", `{"metadata":{"annotations":{"openshift.io/ldap.uid":"cn=garbage"}}}`).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).To(o.ContainSubstring("group.user.openshift.io/secondgroup patched"))
+			out, err = oc.Run("adm", "groups").Args("prune", "--whitelist="+path.Join(currentDir, "ldapgroupuids.txt"), "--sync-config="+syncConfigUserDefinedPath, "--confirm").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).To(o.ContainSubstring("group/secondgroup"))
+			checkOnlyGroupsExist(userClient, "firstgroup", "thirdgroup")
+
+			// cleanup
+			err = userClient.UserV1().Groups().DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// PAGING
+			g.By("Sync all LDAP groups from LDAP server using paged queries")
+			_, err = oc.Run("adm", "groups", "sync").Args("--sync-config="+path.Join(currentDir, "sync-config-paging.yaml"), "--confirm").Output()
+			compareAndCleanup(oc, path.Join(currentDir, "valid_all_ldap_sync.yaml"))
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// SPECIAL SNOWFLAKES
+			if schema == "rfc2307" {
+				// special test for RFC2307
+				g.By("\tTEST: Sync groups from LDAP server, tolerating errors")
+				_, stderr, err := oc.Run("adm", "groups", "sync").Args("--sync-config="+path.Join(currentDir, "sync-config-tolerating.yaml"), "--confirm").Outputs()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				o.Expect(stderr).To(o.ContainSubstring(`For group "cn=group1,ou=groups,ou=incomplete-rfc2307,dc=example,dc=com", ignoring member "cn=INVALID,ou=people,ou=rfc2307,dc=example,dc=com"`))
+				// FIXME: out-of-scope has new logs?
```

Comment:
@sallyom: if my memory serves me well, you were doing some changes to the way scopes are handled in this groups sync, is that so? Could it have something to do with these changed longs?

Reply:
:eyes:

Reply:
https://github.com/openshift/oc/pull/434/files

Reply:
this error is now a log, not an error